element_animal/subject.py

```diff
@@ -146,15 +146,15 @@ class Subject(dj.Manual):
     """Animal subject information.
 
     Attributes:
-        subject ( varchar(8) ): Subject ID.
-        subject_nickname ( varchar(8) ): Subject nickname.
+        subject ( varchar(16) ): Subject ID.
+        subject_nickname ( varchar(64) ): Subject nickname.
         sex (enum): 'M', 'F', or 'U'; Male, Female, or Unknown.
         subject_birth_date (date): Birth date of the subject.
         subject_description ( varchar(1024) ): Description of the subject.
     """
 
     definition = """
-    subject                 : varchar(8)
+    subject                 : varchar(16)
```

Comment:
Is there a good reason to increase to 16.  Since this id percolates into all ids everywhere, we want to force users to come up with a short id for their subjects, also providing a longer nickname.

Reply:
Even varchar(8) is quite long. I would even advocate for a shorter id. Consider that every cell, every trace, etc may have it as part of their id.

Reply:
@ttngu207 and I have run into several cases where the names are longer than 8 characters. Rather than having the labs redo their entire subject naming system, this change will accommodate their use of our element. 

An alternate solution is to fork this repository where this attribute can have more characters but that shifts the burden to maintaining other changes to whoever has created the fork. Therefore @ttngu207 and I decided to propose the change here to open the discussion on this topic.

Reply:
If an ID comes from another source and is not under our control, then we can make it a secondary unique index. 

```
subject :  varchar(8)
---
lims_id : varchar(64)
UNIQUE INDEX (lims_id)
```

Reply:
What would then be ingested into the `subject` field? We will need to come up with a new naming system, and then referencing the "lims_id" - e.g. truncating the `lims_id` (this can't guarantee uniqueness). 
The users would then have to learn to use this new naming scheme, instead of being able to refer to their animals by the original `lims_id`. Or they do a join with `Subject` table for every single one of their queries. 

I think this approach does guarantee best DB design practice but place a burden on usability.

Reply:
Yes, users will have to come up with a way of identifying animals within the database. We can do a hash of the `lims_id`, a sequential number, or random number.

Reply:
we can write a function that suggests a unique identifier from derived from the lims_id.